packages/maestro/src/generator/story.ts

```diff
@@ -0,0 +1,67 @@
+import { StoryObj } from "@storybook/html";
+
+import { code } from "./code";
+import { UIPatternValue } from "./schema";
+
+import { MaestroStory } from ".";
+
+function stories(pattern: UIPatternValue): MaestroStory["stories"] {
+  const props = Object.assign({}, pattern.fields, pattern.settings);
+  const defaultStory: StoryObj = {
+    storyName: "Default",
+    args: {},
+  };
+
+  for (const [key, value] of Object.entries(props)) {
+    defaultStory.args![key] = value.preview;
+  }
+
+  defaultStory.parameters = {
+    docs: {
+      source: {
+        code: code(pattern.use, defaultStory.args!),
+      },
+    },
+  };
+
+  if (!pattern.variants) {
+    return [defaultStory];
+  }
+
+  const stories: StoryObj[] = [defaultStory];
+
+  for (const [key, variant] of Object.entries(pattern.variants)) {
+    if (key === "default") continue;
+    const merged = Object.assign({}, variant.fields, variant.settings);
```

Comment:
You could use the spread syntax instead of Object.assign.

Reply:
I prefer to use `Object.assign` when I'm just merging the objects. Spread is a better tool for modifying or adding a property, but not for merging; `Object.assign` is more declarative and scalable